magicctapipe/scripts/lst1_magic/semi_automatic_scripts/list_from_h5.py

```diff
@@ -0,0 +1,275 @@
+
+        run_listed = []
+        if source_in is None:
+            df_source = df[df["source"] == source_name]
+        else:
+            df_source = df[df["source"] == source_in]
+
+        if is_LST:
+            print("Finding LST runs...")
+            LST_run = df_source["LST1_run"].tolist()  # List with runs as strings
+            LST_date = df_source["date_LST"].tolist()
+            for k in range(len(df_source)):
+                skip = False
+                if np.isnan(LST_run[k]):
+                    continue
+
+                if (int(LST_run[k]) in skip_LST) or (int(LST_run[k]) in run_listed):
+                    skip = True
+
+                if skip is False:
+                    with open(file_list[0], "a+") as f:
+                        f.write(
+                            f"{LST_date[k].replace('-','_')},{str(LST_run[k]).lstrip('0')}\n"
+                        )
+                    run_listed.append(int(LST_run[k]))
+
+        if not is_LST:
+            print("Finding MAGIC runs...")
+            MAGIC_date = df_source["date_MAGIC"].tolist()
+            M2_run = df_source["Run ID"].tolist()
+            for k in range(len(df_source)):
+                skip = False
+                if np.isnan(M2_run[k]):
+                    continue
+
+                if (int(M2_run[k]) in skip_MAGIC) or (int(M2_run[k]) in run_listed):
+                    skip = True
+                if int(M2_run[k]) not in M1_run_list:
+                    skip = True
+
+                if skip is False:
+                    with open(file_list[1], "a+") as f:
+                        f.write(f"{MAGIC_date[k].replace('-','_')},{int(M2_run[k])}\n")
+                    run_listed.append(int(M2_run[k]))
+
+
+def main():
+
+    """
+    This function is automatically called when the script is launched.
+    It calls the functions above to create the files XXXXXX_LST_runs.txt and XXXXX_MAGIC_runs.txt for the desired targets.
+    """
+
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        "--config-file",
+        "-c",
+        dest="config_file",
+        type=str,
+        default="./config_general.yaml",
+        help="Path to a configuration file config_general.yaml",
+    )
+
+    args = parser.parse_args()
+    with open(
+        args.config_file, "rb"
+    ) as f:  # "rb" mode opens the file in binary format for reading
+        config = yaml.safe_load(f)
+
+    source_in = config["data_selection"]["source_name_database"]
+    source_out = config["data_selection"]["source_name_output"]
+    range = config["data_selection"]["time_range"]
+    skip_LST = config["data_selection"]["skip_LST_runs"]
+    skip_MAGIC = config["data_selection"]["skip_MAGIC_runs"]
+
+    df_LST = pd.read_hdf(
+        "/fefs/aswg/workspace/elisa.visentin/auto_MCP_PR/observations_LST.h5",
+        key="joint_obs",
+    )  # TODO: put this file in a shared folder
+
+    df_LST = split_lst_date(df_LST)
+    df_LST = df_LST.astype(
```

Comment:
I tried to run it and got an error because of NANs in LST1_run

Reply:
Now it should be fixed